common/src/main/java/org/dash/wallet/common/services/BlockchainStateProvider.kt

```diff
@@ -0,0 +1,21 @@
+package org.dash.wallet.common.services
+
+import kotlinx.coroutines.flow.Flow
+import org.dash.wallet.common.data.BlockchainState
+
+/**
+ * Blockchain state provider
+ *
+ * This data provider gives access to the current BlockchainState (block height, isSynced)
+ * as well as other data that depends on the current state of the blockchain.
+ *
+ * This includes:
+ *  - Masternode APY (function of block height, difficulty, and network)
+ */
+
+interface BlockchainStateProvider {
+    suspend fun getState(): BlockchainState?
+    fun observeState() : Flow<BlockchainState?>
+    fun getMasternodeAPY(): Double
+    fun getLastMasternodeAPY(): Double
```

Comment:
The return results of these functions will rarely change (every 3 months).

They should only be called when the BlockchainState is synced.